agent/taskresource/types/types.go

```diff
@@ -51,35 +51,43 @@ func (rm *ResourcesMap) UnmarshalJSON(data []byte) error {
 	}
 	result := make(map[string][]taskresource.TaskResource)
 	for key, value := range resources {
-		switch key {
-		case CgroupKey:
-			if unmarshlCgroup(key, value, result) != nil {
-				return err
-			}
-		case DockerVolumeKey:
-			if unmarshalDockerVolume(key, value, result) != nil {
-				return err
-			}
-		case ASMAuthKey:
-			if unmarshalASMAuthKey(key, value, result) != nil {
-				return err
-			}
-		case SSMSecretKey:
-			if unmarshalSSMSecretKey(key, value, result) != nil {
-				return err
-			}
-		case ASMSecretKey:
-			if unmarshalASMSecretKey(key, value, result) != nil {
-				return err
-			}
-		default:
-			return errors.New("Unsupported resource type")
+		if err := unmarshalResource(key, value, result); err != nil {
+			return err
 		}
 	}
 	*rm = result
 	return nil
 }
 
+func unmarshalResource(key string, value json.RawMessage, result map[string][]taskresource.TaskResource) error {
+	switch key {
+	case CgroupKey:
+		if err := unmarshlCgroup(key, value, result); err != nil {
```

Comment:
these could all be simplified to:

return umarshalThing(key,value,result)

```
// this...
if err != nil {
    return err
} else {
    return nil
}

// ...is the same as
return err

Reply:
updated